src/aics-image-viewer/shared/utils/urlParsing.ts

```diff
@@ -21,50 +21,53 @@ const DEFAULT_CONTROL_POINT_COLOR_CODE = "1";
 /** If the `url` query param equals this, parse image url(s) and metadata from local storage instead of params */
 const URL_FETCH_FROM_STORAGE = "storage";
 
-// TODO: refactor regexes to be composed of one another rather than duplicating code
-// const COLOR_CODES: Record<string, ColorArray> = {
-//   "0": [0, 0, 0],
-//   "1": [255, 255, 255],
-//   "-1": [255, 255, 255],
-//   w: [255, 255, 255],
-//   k: [0, 0, 0],
-// };
-// const COLOR_CODE_REGEX = new RegExp(`(${Object.keys(COLOR_CODES).join("|")})`);
-// const HEX_COLOR_REGEX = new RegExp(`(([0-9a-fA-F]{6})|${COLOR_CODE_REGEX.source})`);
+const FLOAT_REGEX = /-?[0-9]*.?[0-9]+/;
 
 const CHANNEL_STATE_KEY_REGEX = /^c[0-9]+$/;
+
 /** Match colon-separated pairs of alphanumeric strings */
 const LUT_REGEX = /^-?[a-z0-9.]*:[ ]*-?[a-z0-9.]*$/;
-/** Match colon-separated pairs of numeric strings */
-const RAMP_REGEX = /^-?[0-9.]*:-?[0-9.]*$/;
+
+/**
+ * Match colon-separated pairs of numeric strings, representing histogram bin
+ * indices or intensity values.
+ */
+const RAMP_REGEX = new RegExp(`${FLOAT_REGEX.source}:${FLOAT_REGEX.source}$`);
+
 /**
  * Match comma-separated triplet of numeric strings.
  */
-const SLICE_REGEX = /^[0-9.]*,[0-9.]*,[0-9.]*$/;
+const SLICE_REGEX = new RegExp(`^${FLOAT_REGEX.source},${FLOAT_REGEX.source},${FLOAT_REGEX.source}$`);
+
 /**
  * Matches a sequence of three comma-separated min:max number pairs, representing
  * the x, y, and z axes.
  */
-const REGION_REGEX = /^([0-9.]*:[0-9.]*)(,[0-9.]*:[0-9.]*){2}$/;
+const REGION_REGEX = new RegExp(
+  `^(${FLOAT_REGEX.source}:${FLOAT_REGEX.source})(,${FLOAT_REGEX.source}:${FLOAT_REGEX.source}){2}$`
+);
 
 const HEX_COLOR_REGEX = new RegExp(`(([0-9a-fA-F]{6})|${DEFAULT_CONTROL_POINT_COLOR_CODE})`);
-const NUMERIC_REGEX = /-?[0-9.]*/;
-const CONTROL_POINT_REGEX = new RegExp(`(${NUMERIC_REGEX.source}:${NUMERIC_REGEX.source}:${HEX_COLOR_REGEX.source})`);
+
+/** Represents control points specified by bin indices. */
+const CONTROL_POINT_REGEX = new RegExp(`(${FLOAT_REGEX.source}:${FLOAT_REGEX.source}:${HEX_COLOR_REGEX.source})`);
```

Comment:
`^` for these ones too? (could be no)

Reply:
Not for these since multiple of them will be chained together